theano/tensor/elemwise.py

```diff
@@ -1993,10 +1999,24 @@ class Sum(CAReduceDtype):
 
     """
 
+    __props__ = ("axis", "dtype", "acc_dtype")
+
     def __init__(self, axis=None, dtype=None, acc_dtype=None):
         CAReduceDtype.__init__(self, scalar.add, axis=axis,
                                dtype=dtype, acc_dtype=acc_dtype)
 
+    def __str__(self):
+        name = self.__class__.__name__
+        axis = ""
+        if self.axis is not None:
+            axis = ", ".join(str(x) for x in self.axis)
+            axis = "axis=[%s], " % axis
+        return "%s{%sacc_dtype=%s}" % (
+            name,
+            axis,
+            str(self.acc_dtype)
+        )
```

Comment:
See the other comment that talk about metaclass. This explain why we need a new \__str__ method here.

As I think it is better that we don't change the output, keep this method.